src/test/java/teammates/test/cases/ui/pagedata/InstructorFeedbackEditPageDataTest.java

```diff
@@ -0,0 +1,330 @@
+        data = new InstructorFeedbackEditPageData(dataBundle.accounts.get("instructor1OfCourse1"));
+        
+        fs = dataBundle.feedbackSessions.get("empty.session");
+        fs.isPublishedEmailEnabled = false;
+        fs.isClosingEmailEnabled = false;
+        
+        questions = new ArrayList<FeedbackQuestionAttributes>();
+        
+        copiableQuestions = new ArrayList<FeedbackQuestionAttributes>();
+        
+        questionHasResponses = new HashMap<String, Boolean>();
+        
+        studentList = new ArrayList<StudentAttributes>();
+        
+        instructorList = new ArrayList<InstructorAttributes>();
+        
+        instructor = getInstructorFromBundle("instructor1OfCourse1");
+        
+        data.init(fs, questions, copiableQuestions, questionHasResponses, studentList, instructorList, instructor);
+        fsForm = data.getFsForm();
+        assertEquals((new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACK_EDIT_COPY_PAGE)
+                          .withUserId(instructor.googleId)).toString(), 
+                      fsForm.getCopyToLink().toString());
+        assertEquals(fs.courseId, fsForm.getCourseId());
+        assertNull(fsForm.getCourses());
+        assertNull(fsForm.getCoursesSelectField());
+        assertFalse(fsForm.isFeedbackSessionTypeEditable());
+        assertTrue(fsForm.isEditFsButtonsVisible());
+        assertNull(fsForm.getFeedbackSessionTypeOptions());
+        
+        additionalSettings = data.getFsForm().getAdditionalSettings();
+        assertEquals(TimeHelper.formatDate(fs.resultsVisibleFromTime), additionalSettings.getResponseVisibleDateValue());
+        assertEquals(TimeHelper.formatDate(fs.sessionVisibleFromTime), additionalSettings.getSessionVisibleDateValue());
+        
+        assertFalse(additionalSettings.isResponseVisiblePublishManuallyChecked());
+        assertTrue(additionalSettings.isResponseVisibleDateChecked());
+        assertFalse(additionalSettings.isResponseVisibleImmediatelyChecked());
+        assertFalse(additionalSettings.isResponseVisibleNeverChecked());
+        assertFalse(additionalSettings.isResponseVisibleDateDisabled());
+       
+        assertFalse(additionalSettings.isSessionVisibleAtOpenChecked());
+        assertFalse(additionalSettings.isSessionVisibleDateDisabled());
+        assertTrue(additionalSettings.isSessionVisibleDateButtonChecked());
+        assertFalse(additionalSettings.isSessionVisiblePrivateChecked());
+        assertFalse(additionalSettings.isSendClosingEmailChecked());
+        assertTrue(additionalSettings.isSendOpeningEmailChecked());
+        assertFalse(additionalSettings.isSendPublishedEmailChecked());
+        
+        
+        questionForms = data.getQnForms();
+        assertEquals(0, questionForms.size());
+        
+        previewForm = data.getPreviewForm();
+        assertEquals(0, previewForm.getStudentToPreviewAsOptions().size());
+        assertEquals(0, previewForm.getInstructorToPreviewAsOptions().size());
```

Comment:
Are these integers gotten from `studentList.size()` and `instructorList.size()`?
It may be better to state them explicitly; it would make for more efficient copy-and-pasting (or can just be extracted into a re-usable method).